src/settings/LoanHistory/utils/normalize.js

```diff
@@ -0,0 +1,48 @@
+import {
+  cloneDeep,
+  forEach,
+  set,
+  isArray,
+  isString,
+} from 'lodash';
+import { change } from 'redux-form';
+
+import { closingTypesMap } from '../../../constants';
+
+const checkInterval = ({ data, dispatch }) => {
+  const newData = cloneDeep(data);
+
+  forEach(Object.entries(data.closingType), ([key, value]) => {
+    if (isString(value) && value !== closingTypesMap.INTERVAL) {
+      set(newData, key, {});
+      dispatch(change('loanHistoryForm', key, {}));
+    } else if (isArray(value)) {
+      forEach(value, (loanExceptionsValue, index) => {
+        if (loanExceptionsValue !== closingTypesMap.INTERVAL) {
+          set(newData, [key, index], {});
+          dispatch(change('loanHistoryForm', [key, index], {}));
+        }
+      });
+    }
+  });
+  return newData;
+};
+
+const filter = (entity, ...callbacks) => {
```

Comment:
I'm not sure that we need to have this function here because its name is a little bit confusing and we can try to do this job in scope of the normalize function.

@VictorSoroka1 What do you think about it?

Reply:
Agree, on top of that in the current approach I do not like that we have two calls of `cloneDeep` function. I would rather not go for it if possible.